pytest/common_lib/shared/backup_service.py

```diff
@@ -0,0 +1,115 @@
+import atexit
+from dataclasses import dataclass, field
+import json
+import os
+import shutil
+import subprocess
+import tempfile
+from typing import Any, cast
+
+import base58
+from common_lib.constants import BACKUP_SERVICE_BINARY_PATH, MPC_REPO_DIR
+from common_lib.migration_state import BackupServiceInfo
+from common_lib.shared.mpc_node import MpcNode
+
+from nacl.signing import SigningKey
+
+BackupServiceSecretsJson = dict[str, list[int]]
+
+
+@dataclass
+class BackupServiceSecrets:
+    p2p_private_key: bytes
+    near_signer_key: bytes
+    local_storage_aes_key: bytes
+
+    @staticmethod
+    def from_dict(backup_json: BackupServiceSecretsJson) -> "BackupServiceSecrets":
+        return BackupServiceSecrets(
+            p2p_private_key=bytes(backup_json["p2p_private_key"]),
+            near_signer_key=bytes(backup_json["near_signer_key"]),
+            local_storage_aes_key=bytes(backup_json["local_storage_aes_key"]),
+        )
+
+    def near_public_key(self) -> str:
+        sk = SigningKey(self.p2p_private_key)
+        pk_bytes = sk.verify_key.encode()
+        pk_b58 = base58.b58encode(pk_bytes).decode()
+        return f"ed25519:{pk_b58}"
+
+
+@dataclass
+class BackupService:
+    home_dir: str = field(init=False)
+
+    def __post_init__(self):
```

Comment:
Let's use a normal constructor here:

```suggestion
    def __init__(self):
```